File: sdks/csharp/src/PogodocApi/Templates/Requests/UpdateTemplateRequest.cs
using System.Text.Json.Serialization;
using PogodocApi;

#nullable enable

namespace PogodocApi;

public record UpdateTemplateRequest
{
    [JsonPropertyName("templateInfo")]
    public required UpdateTemplateRequestTemplateInfo TemplateInfo { get; init; }

    [JsonPropertyName("previewIds")]
    public required UpdateTemplateRequestPreviewIds PreviewIds { get; init; }

    [JsonPropertyName("contentId")]
    public required string ContentId { get; init; }
}

File: sdks/php/src/Templates/Requests/UpdateTemplateRequest.php
<?php

namespace Pogodoc\Templates\Requests;

use Pogodoc\Templates\Types\UpdateTemplateRequestTemplateInfo;
use Pogodoc\Core\JsonProperty;
use Pogodoc\Templates\Types\UpdateTemplateRequestPreviewIds;

class UpdateTemplateRequest
{
    /**
     * @var UpdateTemplateRequestTemplateInfo $templateInfo
     */
    #[JsonProperty('templateInfo')]
    public UpdateTemplateRequestTemplateInfo $templateInfo;

    /**
     * @var UpdateTemplateRequestPreviewIds $previewIds
     */
    #[JsonProperty('previewIds')]
    public UpdateTemplateRequestPreviewIds $previewIds;

    /**
     * @var string $contentId
     */
    #[JsonProperty('contentId')]
    public string $contentId;

    /**
     * @param array{
     *   templateInfo: UpdateTemplateRequestTemplateInfo,
     *   previewIds: UpdateTemplateRequestPreviewIds,
     *   contentId: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->templateInfo = $values['templateInfo'];
        $this->previewIds = $values['previewIds'];
        $this->contentId = $values['contentId'];
    }
}
